### PR TITLE
gitignore: add `/buck-out/` to ignore list

### PR DESCRIPTION
**Pull request by thoughtpolice**
While working on the Buck branch, I have been bitten about a million times by the following setup:

- Hack away
- Switch back to main
- Run `jj st` after editing a file
- `/buck-out` gets snapshotted, because it isn't ignored
- Trip max snapshot filesize error
- Pain

There are a few ways to skin the cat but this is simplest. It's an incredibly small addition and it may be all for naught in the long run, but this will stem the bleeding, at least.

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [ ] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

**Comment by istudyatuni** [comment]
You can just add to `.git/info/exclude`

**Comment by thoughtpolice** [comment]
> You can just add to `.git/info/exclude`

Yes, but on every single repository I have to clone on all my machines everywhere, forever. Please see the previous discussion in https://github.com/martinvonz/jj/pull/4409 &mdash; I will admit don't like this solution, to a degree that is almost inexplicable (why add a thing this branch doesn't need, I guess?) but at the end of the day I want to *not* snapshot thousands of files on accident, every other day, in a foolproof way, and this change perhaps is uncontroversial.

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /target/
+/buck-out/
```
Comment (ilyagr) [inline .gitignore:2]: I would put this on the bottom and add a comment: `# To make working on ... easier`, perhaps with an issue id.

**Review by ilyagr — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (thoughtpolice) [reply .gitignore]: Done. I also went ahead and made the capitalization in comments here consistent.

**Pull request merged**